src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/CompilerServices/Operators.vb

```diff
@@ -4610,27 +4591,18 @@ Namespace Microsoft.VisualBasic.CompilerServices
         End Function
 
         Private Shared Function ModInt16(ByVal left As Int16, ByVal right As Int16) As Object
-            Dim result As Integer = CInt(left) Mod CInt(right)
-
-            If result < Int16.MinValue OrElse result > Int16.MaxValue Then
-                Return result
-            Else
-                Return CShort(result)
-            End If
+            Return left Mod right
```

Comment:
This can't overflow, and the degenerate case, if `left == Int16.MinValue` and `right == -1` does not overflow and returns 0